### PR TITLE
Update chpldoc to work when CHPL_HOME is moved or changed.

### PR DESCRIPTION
**Pull request by thomasvandoren**
Previously, the virtualenv install for the python packages hard coded the paths
at the time of install. If this was built into the module rpm, for example, the
path to CHPL_HOME would change potentially with each install of the module and
chpldoc would not work.

Update the python package install to use virtualenv --relocatable [1], which
updates all the hard coded paths to be relative. One of the caveats when using
the --relocatable flag, is the activate scripts no longer work. Fortunately,
the only required things the activate scripts do is set VIRTUAL_ENV in the
environment and update PATH to have the python "binaries" from the virtualenv
at the beginning. So, also update the docs() pass to set VIRTUAL_ENV and PATH
environment variables instead of calling `. path/to/activate`.

I verified this by copying my chapel working directory with a build of chpldoc
somewhere else on my file system, deleting the install dir in
third-party/chpldoc-venv/ from the original workspace, and then running chpldoc
in the new directory. It worked as expected for `make docs`.

Also, fix spelling error in name for virtualenv directory (chpdoc-virtualenv ->
chp**l**doc-virtualenv).

1: https://virtualenv.pypa.io/en/latest/userguide.html#making-environments-relocatable

### CODE REVIEW

**Comment by thomasvandoren** [comment]
@lydia-duncan -- Will you please review?

**Comment by thomasvandoren** [comment]
Just to make sure, I ran start_test over test/release/examples/primers/chpldoc.chpl and test/chpldoc/.

File: compiler/passes/docs.cpp [hunk 1]
```diff
@@ -272,20 +272,22 @@ std::string generateSphinxProject(std::string dirpath) {
  * outputDir for generated html files.
  */
 void generateSphinxOutput(std::string sphinxDir, std::string outputDir) {
-  // The virtualenv active and sphinx-build scripts are in:
-  //   $CHPL_HOME/third-party/chpldoc-venv/install/$CHPL_TARGET_PLATFORM/chpldoc-virtualenv/bin/
-  const char * venvBinDir = astr(
+  // Set the PATH and VIRTUAL_ENV variables in the environment. The values are
+  // based on the install path in third-party/chpldoc-venv/ dir.
```
Comment (lydia-duncan) [inline compiler/passes/docs.cpp]: "in *the* third-party/chpldoc-venv/ dir"?

**Comment by lydia-duncan** [comment]
Looks great, thanks!

**Pull request merged**